source/clear-linux/tutorials/docker/docker.rst

```diff
@@ -99,13 +99,20 @@ exist by default.
 
    .. code-block:: bash
 
-      touch /etc/docker/daemon.json
+      sudo touch /etc/docker/daemon.json
 
    .. note:: 
 
       Refer to the `Docker documentation on daemon configuration`_ for the 
       full list of available configuration options and examples.
 
+      An minimal configuration would be.
```

Comment:
@odwdinc are you saying this storage-driver change was required for your deployment to work or that this is an example of an overriding change you can make?

If the first, I'd be curious as to what issues you experienced since I didn't run into it.
If the latter, perhaps rephrase to better describe what is being doing and indicate it's an example/override/optional.

Reply:
yes it was need for me to get, docker pull working.
let me confirm it still needed.

Reply:
In reading Docker's docs, I notice the daemon `storage driver options` are spelled out pretty well. 
https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-storage-driver
https://docs.docker.com/engine/reference/commandline/dockerd/#options-per-storage-driver

Yet it's not clear to me --and I'd appreciate your explanation-- why this would be considered a minimal configuration on CL? Per Docker's docs, would a "minimal" configuration also require another key whose values are stored in a list--as *Options per storage driver* shows?

Reply:
@odwdinc thanks for explaining.

My suspicion is that the storage driver change was probably not the root cause of the `docker pull` problem you had. Let us know what your testing concludes.

Reply:
Agreed.  Please report back @odwdinc

Reply:
so im still geting this
sudo systemctl restart docker
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
journalctl -xe

 `dockerd[22086]: unable to configure the Docker daemon with file /etc/docker/daemon.json: EOF
systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: docker.service: Failed with result 'exit-code'.
systemd[1]: Failed to start Docker Application Container Engine.
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.service has failed.
`

when just running just sudo touch /etc/docker/daemon.json
adding the storage-driver  bit allows me to restart docker.

Reply:
Can you `rm -rfv /etc/docker` and then restart the docker.service to test? 

I have another system here without a daemon.json working so it's odd. Another thing that makes this behavior odd is it would be against the stateless and sane defaulting nature of Clear.

`sudo docker info` and `sudo swupd info` and `swupd bundle-list` would be good to know too.

Reply:
sudo rm -rfv /etc/docker
Password:
removed '/etc/docker/daemon.json'
removed '/etc/docker/key.json'
removed directory '/etc/docker'
epray@Elloitt ~ $ sudo systemctl restart docker
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

 sudo docker info
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

sudo swupd info
Installed version: 25720
Version URL:       https://download.clearlinux.org/update/
Content URL:       https://cdn.download.clearlinux.org/update/

swupd bundle-list
bootloader
containers-basic
desktop
desktop-apps
desktop-assets
desktop-autostart
desktop-gnomelibs
desktop-locales
dev-utils
editors
emacs
ethtool
games
git
go-basic
go-basic-dev
iproute2
joe
kernel-native
kvm-host
libX11client
mail-utils
network-basic
openssh-server
openssl
os-core
os-core-dev
os-core-update
p11-kit
perl-basic
python2-basic
python3-basic
shells
storage-utils
storage-utils-dev
sudo
sysadmin-basic
telemetrics
user-basic
vim
which
wpa_supplicant
x11-server
xemacs

Reply:
Paging @bryteise - any thoughts on the issue of docker really wanting `/etc/docker/daemon.json`?

Reply:
updated to 25800 still not able to run 
sudo systemctl restart docker
Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

journalctl -xe
systemd[1]: docker.service: Service RestartSec=100ms e>
systemd[1]: docker.service: Scheduled restart job, res>
-- Subject: Automatic restarting of a unit has been scheduled
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Automatic restarting of the unit docker.service has been scheduled, as the >
-- the configured Restart= setting for the unit.
systemd[1]: Stopped Docker Application Container Engin>
-- Subject: Unit docker.service has finished shutting down
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.service has finished shutting down.
systemd[1]: docker.service: Start request repeated too>
systemd[1]: docker.service: Failed with result 'exit-c>
systemd[1]: Failed to start Docker Application Contain>
-- Subject: Unit docker.service has failed
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit docker.service has failed.
--
-- The result is RESULT.

@puneetse how are you running Clear Linux, aka kvm, hyperv,etc.. i'm ruining on bar metal.

Reply:
@odwdinc Currently 2 VMs on different platforms and 1 baremetal.

Reply:
more info on this,
it seem to work if you  never create the file, /etc/docker/daemon.json
once the file is created the error will not go away with the changes added to the file.
even removing the file will not undo what happens when adding.